mediathread/templates/projects/composition_assignment.html

```diff
@@ -120,22 +120,39 @@ <h1 class="page-title">
                         <div class="dropdown d-inline ml-1 response-dropdown">
                             <button class="btn btn-primary dropdown-toggle" type="button" id="student-response-dropdown"
                                 data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-                                {{responses|length}} of {{students|length}} students responded
+                                {% get_submitted responses as submitted %}
+                                {{submitted|length}} of {{students|length}} Students Responded
                             </button>
                             <div class="dropdown-menu dropdown-menu-right" aria-labelledby="student-response-dropdown">
-                                {% for student in students %}
-                                    {% student_response responses student as response %}
-                                    {% if response.date_submitted %}
-                                        <a href="{% url 'project-workspace' request.course.id response.id %}"
-                                            class="dropdown-item" type="button">
-                                            {% public_name for student %} ({{response.date_submitted|date:"m/d/Y"}})
-                                        </a>
-                                    {% else %}
-                                        <a href="#" class="dropdown-item disabled" type="button">
-                                            {% public_name for student %}
-                                        </a>
-                                    {% endif %}
-                                {% endfor %}
+                                <h6 class="dropdown-header">Submitted
+                                    {% for student in students %}
```

Comment:
Rather than iterating students here...you might be able to iterate `submitted`. This could make the page a bit peppier as getting a student's response & feedback can be heavy operations.

Reply:
@sdreher right! Okay I refactored a bit so it's not so heavy.